smoke/IotEdgeQuickstart/details/IotedgedWindows.cs

```diff
@@ -52,24 +54,61 @@ public async Task VerifyModuleIsRunning(string name)
                 string errorMessage = null;
+                int retryCount = 5;
 
                 try
                 {
                     while (true)
                     {
-                        await Task.Delay(TimeSpan.FromSeconds(3), cts.Token);
-
-                        string[] result = await Process.RunAsync("iotedge", "list", cts.Token);
-                        WriteToConsole("Output of iotedge list", result);
-
-                        string status = result
-                            .Where(ln => ln.Split(null as char[], StringSplitOptions.RemoveEmptyEntries).First() == name)
-                            .DefaultIfEmpty("name status")
-                            .Single()
-                            .Split(null as char[], StringSplitOptions.RemoveEmptyEntries)
-                            .ElementAt(1);  // second column is STATUS
-
-                        if (status == "running") break;
+                        string status = string.Empty;
+
+                        try
+                        {
+                            string[] result = await Process.RunAsync("iotedge", "list", cts.Token);
+                            WriteToConsole("Output of iotedge list", result);
+
+                            status = result
+                                .Where(ln => ln.Split(null as char[], StringSplitOptions.RemoveEmptyEntries).First() == name)
+                                .DefaultIfEmpty("name status")
+                                .Single()
+                                .Split(null as char[], StringSplitOptions.RemoveEmptyEntries)
+                                .ElementAt(1); // second column is STATUS
+
+                            if (status == "running")
+                            {
+                                break;
+                            }
+                            else
+                            {
+                                await Task.Delay(TimeSpan.FromSeconds(5), cts.Token);
+                            }
+                        }
+                        catch (Exception listOperationException)
+                        {
+                            // Display error and retry for some transient exceptions such as hyper error
+                            string exceptionDetails = listOperationException.ToString();
+                            Console.WriteLine(exceptionDetails);
+                            retryCount--;
+
+                            if (retryCount < 0)
+                            {
+                                throw;
+                            }
+
+                            if (exceptionDetails.Contains("Could not list modules", StringComparison.OrdinalIgnoreCase))
+                            {
+                                try
+                                {
+                                    Console.WriteLine("Workaround: restart iotedge service.");
+                                    await this.Restart().ConfigureAwait(true);
```

Comment:
Is this ConfigureAwait(true) needed? I would avoid it if not..

Reply:
Updated in another PR.